UNITTESTS/stubs/CMakeLists.txt

```diff
@@ -115,7 +105,6 @@ target_include_directories(mbed-stubs-headers
 add_subdirectory(connectivity)
 add_subdirectory(events)
 add_subdirectory(hal)
-add_subdirectory(platform)
 add_subdirectory(storage)
 
 add_library(mbed-stubs INTERFACE)
```

Comment:
(Referring to the lines below)
Shall we remove mbed-stubs-platform from mbed-stubs, and let tests explicitly link mbed-stubs-platform? Similar to what's been done to the headers - only link what is used.

Reply:
I recall some build failures when I tried that initially. I think we can only remove this when everything else depending on platform has been refactored. I'll take another look and see how tricky it is to untangle at this stage, now that we've merged some other stubs PRs.

Reply:
I think if we take it out from mbed-stubs, lots of tests need to have mbed-stubs-platform explicitly added. I imagine platform is the trickiest given everything depends on it, and it might be true for hal too.